pandas/tseries/holiday.py

```diff
@@ -328,6 +340,12 @@ def dates(
         holiday_dates = holiday_dates[
             (holiday_dates >= filter_start_date) & (holiday_dates <= filter_end_date)
         ]
+
+        if self.exclude_dates:
+            holiday_dates = DatetimeIndex(
+                [hd for hd in holiday_dates if hd not in self.exclude_dates]
+            )
```

Comment:
1. Conversion to `DatetimeIndex` should be done in `__init__`
2. Probably better to use `.difference` or releated method instead